### ISSUE TITLE
AttributeError: PANet: ResNet: 'super' object has no attribute '_specify_ddp_gpu_num'

### ISSUE THREAD

**Issue by mobassir94** [open-mmlab/mmocr#325]
i tried all the detection models of this repo,i was able to run almost all of them except four of them,and four of those detector produce same/similar error.
the 4 py file that is not working for me
1.panet_r18_fpem_ffm_600e_icdar2015.py
2.panet_r18_fpem_ffm_600e_ctw1500.py
3.psenet_r50_fpnf_600e_icdar2015.py
4.psenet_r50_fpnf_600e_ctw1500.py 

i am attaching the config and error log that i get while running panet_r18_fpem_ffm_600e_icdar2015.py below : 

```
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-apsisdev'
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-apsisdev'
Config:
optimizer = dict(type='Adam', lr=0.000125)
optimizer_config = dict(grad_clip=None)
lr_config = dict(policy='poly', power=0.9, warmup=None)
total_epochs = 600
checkpoint_config = dict(interval=10)
log_config = dict(interval=1, hooks=[dict(type='TextLoggerHook')])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
model = dict(
    type='PANet',
    pretrained='torchvision://resnet18',
    backbone=dict(
        type='ResNet',
        depth=18,
        num_stages=4,
        out_indices=(0, 1, 2, 3),
        frozen_stages=-1,
        norm_cfg=dict(type='SyncBN', requires_grad=True),
        norm_eval=True,
        style='caffe'),
    neck=dict(type='FPEM_FFM', in_channels=[64, 128, 256, 512]),
    bbox_head=dict(
        type='PANHead',
        text_repr_type='quad',
        in_channels=[128, 128, 128, 128],
        out_channels=6,
        loss=dict(type='PANLoss')),
    train_cfg=None,
    test_cfg=None)
dataset_type = 'IcdarDataset'
data_root = '/home/apsisdev/IMPORTANT/totaltext'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='LoadTextAnnotations',
        with_bbox=True,
        with_mask=True,
        poly2mask=False),
    dict(type='ColorJitter', brightness=0.12549019607843137, saturation=0.5),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(
        type='ScaleAspectJitter',
        img_scale=[(3000, 736)],
        ratio_range=(0.7, 1.3),
        aspect_ratio_range=(0.9, 1.1),
        multiscale_mode='value',
        keep_ratio=False),
    dict(type='PANetTargets', shrink_ratio=(1.0, 0.5), max_shrink=20),
    dict(type='RandomFlip', flip_ratio=0.5, direction='horizontal'),
    dict(type='RandomRotateTextDet'),
    dict(
        type='RandomCropInstances',
        target_size=(736, 736),
        instance_key='gt_kernels'),
    dict(type='Pad', size_divisor=32),
    dict(
        type='CustomFormatBundle',
        keys=['gt_kernels', 'gt_mask'],
        visualize=dict(flag=False, boundary_key='gt_kernels')),
    dict(type='Collect', keys=['img', 'gt_kernels', 'gt_mask'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(1333, 736),
        flip=False,
        transforms=[
            dict(type='Resize', img_scale=(1333, 736), keep_ratio=True),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size_divisor=32),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=8,
    workers_per_gpu=2,
    train=dict(
        type='IcdarDataset',
        ann_file='/home/apsisdev/IMPORTANT/totaltext/instances_training.json',
        img_prefix='/home/apsisdev/IMPORTANT/totaltext/imgs',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='LoadTextAnnotations',
                with_bbox=True,
                with_mask=True,
                poly2mask=False),
            dict(
                type='ColorJitter',
                brightness=0.12549019607843137,
                saturation=0.5),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(
                type='ScaleAspectJitter',
                img_scale=[(3000, 736)],
                ratio_range=(0.7, 1.3),
                aspect_ratio_range=(0.9, 1.1),
                multiscale_mode='value',
                keep_ratio=False),
            dict(type='PANetTargets', shrink_ratio=(1.0, 0.5), max_shrink=20),
            dict(type='RandomFlip', flip_ratio=0.5, direction='horizontal'),
            dict(type='RandomRotateTextDet'),
            dict(
                type='RandomCropInstances',
                target_size=(736, 736),
                instance_key='gt_kernels'),
            dict(type='Pad', size_divisor=32),
            dict(
                type='CustomFormatBundle',
                keys=['gt_kernels', 'gt_mask'],
                visualize=dict(flag=False, boundary_key='gt_kernels')),
            dict(type='Collect', keys=['img', 'gt_kernels', 'gt_mask'])
        ]),
    val=dict(
        type='IcdarDataset',
        ann_file='/home/apsisdev/IMPORTANT/totaltext/instances_test.json',
        img_prefix='/home/apsisdev/IMPORTANT/totaltext/imgs',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1333, 736),
                flip=False,
                transforms=[
                    dict(
                        type='Resize', img_scale=(1333, 736), keep_ratio=True),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='IcdarDataset',
        ann_file='/home/apsisdev/IMPORTANT/totaltext/instances_test.json',
        img_prefix='/home/apsisdev/IMPORTANT/totaltext/imgs',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(1333, 736),
                flip=False,
                transforms=[
                    dict(
                        type='Resize', img_scale=(1333, 736), keep_ratio=True),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='Pad', size_divisor=32),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
evaluation = dict(interval=10, metric='hmean-iou')
work_dir = '../outputs/detection_model'
seed = 0
gpu_ids = range(0, 1)

loading annotations into memory...
Done (t=0.43s)
creating index...
index created!
Traceback (most recent call last):
  File "/home/apsisdev/mmcv/mmcv/utils/registry.py", line 51, in build_from_cfg
    return obj_cls(**args)
  File "/home/apsisdev/anaconda3/envs/ocr/lib/python3.8/site-packages/mmdet/models/backbones/resnet.py", line 417, in __init__
    self._make_stem_layer(in_channels, stem_channels)
  File "/home/apsisdev/anaconda3/envs/ocr/lib/python3.8/site-packages/mmdet/models/backbones/resnet.py", line 566, in _make_stem_layer
    self.norm1_name, norm1 = build_norm_layer(
  File "/home/apsisdev/mmcv/mmcv/cnn/bricks/norm.py", line 110, in build_norm_layer
    layer._specify_ddp_gpu_num(1)
  File "/home/apsisdev/mmcv/mmcv/utils/parrots_wrapper.py", line 87, in _specify_ddp_gpu_num
    super()._specify_ddp_gpu_num(gpu_size)
AttributeError: 'super' object has no attribute '_specify_ddp_gpu_num'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/apsisdev/mmcv/mmcv/utils/registry.py", line 51, in build_from_cfg
    return obj_cls(**args)
  File "/home/apsisdev/IMPORTANT/mmocr/mmocr/models/textdet/detectors/panet.py", line 24, in __init__
    SingleStageTextDetector.__init__(self, backbone, neck, bbox_head,
  File "/home/apsisdev/IMPORTANT/mmocr/mmocr/models/textdet/detectors/single_stage_text_detector.py", line 19, in __init__
    SingleStageDetector.__init__(self, backbone, neck, bbox_head,
  File "/home/apsisdev/anaconda3/envs/ocr/lib/python3.8/site-packages/mmdet/models/detectors/single_stage.py", line 25, in __init__
    self.backbone = build_backbone(backbone)
  File "/home/apsisdev/anaconda3/envs/ocr/lib/python3.8/site-packages/mmdet/models/builder.py", line 39, in build_backbone
    return build(cfg, BACKBONES)
  File "/home/apsisdev/anaconda3/envs/ocr/lib/python3.8/site-packages/mmdet/models/builder.py", line 34, in build
    return build_from_cfg(cfg, registry, default_args)
  File "/home/apsisdev/mmcv/mmcv/utils/registry.py", line 54, in build_from_cfg
    raise type(e)(f'{obj_cls.__name__}: {e}')
AttributeError: ResNet: 'super' object has no attribute '_specify_ddp_gpu_num'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "train_any_mmocr_detector.py", line 123, in <module>
    model = build_detector(
  File "/home/apsisdev/anaconda3/envs/ocr/lib/python3.8/site-packages/mmdet/models/builder.py", line 77, in build_detector
    return build(cfg, DETECTORS, dict(train_cfg=train_cfg, test_cfg=test_cfg))
  File "/home/apsisdev/anaconda3/envs/ocr/lib/python3.8/site-packages/mmdet/models/builder.py", line 34, in build
    return build_from_cfg(cfg, registry, default_args)
  File "/home/apsisdev/mmcv/mmcv/utils/registry.py", line 54, in build_from_cfg
    raise type(e)(f'{obj_cls.__name__}: {e}')
AttributeError: PANet: ResNet: 'super' object has no attribute '_specify_ddp_gpu_num'

```

all other 3 py file of my list also gave me this same error saying **AttributeError: PANet: ResNet: 'super' object has no attribute '_specify_ddp_gpu_num'**

**Comment by cuhk-hbsun** [comment]
please run `python mmocr/utils/collect_env.py` and paste the information here. Did you install `mmdet==2.11.0` and `mmcv-full==1.3.4` as mentioned in [install_guide](https://mmocr.readthedocs.io/en/latest/install.html#full-set-up-script)?

**Comment by mobassir94** [comment]
dear @cuhk-hbsun 

sys.platform: linux
Python: 3.8.5 (default, Sep  4 2020, 07:30:14) [GCC 7.3.0]
CUDA available: True
GPU 0: GeForce RTX 3090
CUDA_HOME: /usr/local/cuda
NVCC: Build cuda_11.2.r11.2/compiler.29618528_0
GCC: gcc (Ubuntu 9.4.0-1ubuntu1~20.04) 9.4.0
PyTorch: 1.10.0.dev20210623+cu111
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) Math Kernel Library Version 2020.0.0 Product Build 20191122 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.1
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86
  - CuDNN 8.0.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.1, CUDNN_VERSION=8.0.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.0, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.11.0.dev20210623+cu111
OpenCV: 4.5.2
MMCV: 1.3.7
MMCV Compiler: GCC 9.4
MMCV CUDA Compiler: 11.2
MMOCR: 0.2.0+271129f



please check the 2nd last comment of this issue : https://github.com/open-mmlab/mmocr/issues/314
you will understand why i had to upgrade mmcv-full

**Comment by innerlee** [comment]
The compatibility with pytorch 1.10 is not checked yet. Please wait for updates from mmcv.
You can try pytorch 1.7/1.8

**Comment by mobassir94** [comment]
torch 1.7/1.8 had issue with rtx 3090 so i had to upgrade,please check your co workers comments here : https://github.com/open-mmlab/mmocr/issues/314

**Comment by botastark** [comment]
had the same issue! 
resolved after changing type to 'BN' in  norm_cfg=dict(type='SyncBN', requires_grad=True),

**Comment by gaotongxiao** [comment]
Thanks for reporting that!

It seems `_specify_ddp_gpu_num()` is no longer available after PyTorch 1.9.0 and SyncBatchNorm would break. I will raise an issue to mmcv. Before it gets fixed by our colleagues, you may either try @botastark 's solution or create another environment for the elder version of mmcv, though certain architectures may not be fully supported.

**Comment by mobassir94** [comment]
@botastark 's solution worked for me,thank you
